### ISSUE TITLE
TODO: Add title

### ISSUE THREAD

**Issue by Christopherrmm** [koalaman/shellcheck#1973]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2045, SC2035, SC2062, SC2039, SC2016
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
## Example: a typical script with several problems
for f in $(ls *.m3u)
do
  grep -qi hq.*mp3 "$f" \
    && echo -e 'Playlist $f contains a HQ file in mp3 format'
done

```

#### Here's what shellcheck currently says:
```
Line 3	SC2045: Iterating over ls output is fragile. Use globs.
Line 3	SC2035: Use ./*glob* or -- *glob* so names with dashes won't become options.
Line 5	SC2062: Quote the grep pattern so the shell won't interpret it.
Line 6	SC2039: In POSIX sh, echo flags are undefined.
Line 6	SC2016: Expressions don't expand in single quotes, use double quotes for that.
```
#### Here's what I wanted or expected to see:

TODO: Describe expected/desired output

**Comment by koalaman** [comment]
Assuming this was accidentally submitted
